pkgs/applications/terminal-emulators/mlterm/default.nix

```diff
@@ -19,25 +87,44 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-gfs5cdwUUwSBWwJJSaxrQGWJvLkI27RMlk5QvDALEDg=";
   };
 
-  nativeBuildInputs = [ pkg-config autoconf wrapGAppsHook ];
+  nativeBuildInputs = [
+    pkg-config
+    autoconf
+  ] ++ addIfHas "mlconfig" toolsChoices [
```

Comment:
Something tells me this is not the best way to do this. 
It looks like searching a hash table.

Reply:
I'd prefer individual boolean flags rather than list flags. Lists don't compose very well with overrides.

Reply:
> I'd prefer individual boolean flags rather than list flags. Lists don't compose very well with overrides.

Composing overrides indeed that would have been ideal, but these lists are not that long, and users using them should anyway read the source code, and choose the list they want to be compiled.

> Something tells me this is not the best way to do this. It looks like searching a hash table.

I'm open to suggestions. Unless you'd both insist that it's better to use boolean flags.

Reply:
Discussion on a third alternative which you already implemented: https://github.com/NixOS/nixpkgs/pull/226218#discussion_r1167433115

Reply:
I suggest inline this, using `lib.elem x list`

Reply:
> I suggest inline this, using `lib.elem x list`

That was I missing, thanks!